Foundation/JSONSerialization.swift

```diff
@@ -577,9 +577,12 @@ private struct JSONWriter {
             writer(" ")
         }
     }
-    
+
     //[SR-2151] https://bugs.swift.org/browse/SR-2151
     private mutating func _serializationString(for number: NSNumber) -> String {
+        if _CFNumberGetType2(number._cfObject) == kCFNumberSInt128Type {
```

Comment:
Since the `stringValue` can be used for any integer value, I could replace it with: 
```
        if !CFNumberIsFloatType(number._cfObject) {
            return number.stringValue
        }
```